src/Adapters/AzureManagementAdapter.ts

```diff
@@ -256,166 +254,221 @@ export default class AzureManagementAdapter implements IAzureManagementAdapter {
+                    error
+                );
+            }
+
+            return new AzureResourceData(resource);
         }, 'azureManagement');
     }
 
-    /** Returns list of all the resources of the given type in all of user's subscriptions */
-    async getResources({
-        resourceType,
-        searchParams,
-        resourceProviderEndpoint,
-        userData
-    }: AdapterMethodParamsForGetAzureResources) {
+    /** Returns list of all the resources of the given type and access role ids */
+    async getResources(params: AdapterMethodParamsForGetAzureResources) {
         const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
 
-        if (userData) {
-            if (userData.tenantId) {
-                this.tenantId = userData.tenantId;
-            }
-            if (userData.uniqueObjectId) {
-                this.uniqueObjectId = userData.uniqueObjectId;
-            }
-        }
-
-        return await adapterMethodSandbox.safelyFetchData(async () => {
+        return await adapterMethodSandbox.safelyFetchData(async (token) => {
             let resources: Array<IAzureResource> = [];
             try {
-                const subscriptions = await this.getSubscriptions();
-                const userSubscriptions: Array<IAzureSubscription> = subscriptions.getData();
+                logDebugConsole(
+                    'debug',
+                    `[START] Fetching ${params.resourceType} type resources`
+                );
 
                 if (
-                    resourceType === AzureResourceTypes.StorageBlobContainer &&
-                    searchParams?.additionalParams &&
-                    'storageAccountId' in searchParams?.additionalParams
+                    params.resourceType ===
+                    AzureResourceTypes.StorageBlobContainer
                 ) {
-                    if (searchParams?.additionalParams?.storageAccountId) {
-                        const resourcesResponse = await this.getContainersInStorageAccount(
-                            searchParams.additionalParams.storageAccountId
+                    if (
+                        params.searchParams?.additionalParams
+                            ?.storageAccountId ||
+                        params.searchParams?.additionalParams
+                            ?.storageAccountBlobUrl
+                    ) {
+                        const storageAccounts: Array<IAzureStorageAccount> = await this.fetchAllResources(
+                            adapterMethodSandbox,
+                            token,
+                            {
+                                type: AzureResourceTypes.StorageAccount,
+                                query: params.searchParams?.additionalParams
+                                    ?.storageAccountId
+                                    ? "id == '" +
+                                      params.searchParams?.additionalParams
```

Comment:
I personally find it more readable when we use the other notation for building strings with the ` marks and the ${variable}